openedx_learning/core/publishing/models.py

```diff
@@ -1,48 +1,340 @@
+    # user is later removed. Open edX in general doesn't let you remove users,
+    # but we should try to model it so that this is possible eventually.
+    created_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+    )
+
+    class Meta:
+        constraints = [
+            # Prevent the situation where we have multiple
+            # PublishableEntityVersions claiming to be the same version_num for
+            # a given PublishableEntity. This can happen if there's a race
+            # condition between concurrent editors in different browsers,
+            # working on the same Publishable. With this constraint, one of
+            # those processes will raise an IntegrityError.
+            models.UniqueConstraint(
+                fields=[
+                    "entity",
+                    "version_num",
+                ],
+                name="pv_uniq_entity_version_num",
+            )
+        ]
+        indexes = [
+            # LearningPackage (reverse) Created Index:
+            #   * Make it cheap to find the most recently created
+            #     PublishableEntityVersions for a given LearningPackage. This
+            #     represents the most recently saved work for a LearningPackage
+            #     and would be the most likely areas to get worked on next.
+            models.Index(
+                fields=["entity", "-created"],
+                name="pv_idx_entity_rcreated",
+            ),
+            # Title Index:
+            #   * Search by title.
+            models.Index(
+                fields=[
+                    "title",
+                ],
+                name="pv_idx_title",
+            ),
+        ]
+
+        # These are for the Django Admin UI.
+        verbose_name = "Publishable Entity Version"
+        verbose_name_plural = "Publishable Entity Versions"
+
+
+class Draft(models.Model):
     """
-    This model tracks Publishing activity.
+    Find the latest version of any given entity.
+    
+    This model mostly only exists to allow us to join against a bunch of
+    PublishableEntity objects at once and get all their latest drafts. You might
+    use this together with Published in order to see which Drafts haven't been
+    published yet.
+
+    A Draft entry should be created whenever a new PublishableEntityVersion is
+    created. This means there are three possible states:
 
-    It is expected that other apps make foreign keys to this table to mark when
-    their content gets published. This is to allow us to tie together many
-    different entities (e.g. Components, Units, etc.) that are all published at
-    the same time.
+    1. No Draft entry for a PublishableEntity: This means a PublishableEntity
+       was created, but no PublishableEntityVersion was ever made for it, so
+       there was never a Draft version.
+    2. A Draft entry exists and points to a PublishableEntityVersion: This is
```

Comment:
Is this the state when the `PublishableEntityVersion` is also the latest published one?

Reply:
Not necessarily. The Draft entry will always point to the most recently created `PublishableEntityVersion`. That may or may not be the published version–that depends on the entry in the `Published` model.